### PR TITLE
apply maxmsgsize as global domain limit and some smaller fixes

### PR DESCRIPTION
**Pull request by Udera**
After some testing, I found:

A global maxmsgsize was not correctly applied.
User couldn't change his preferences (only password). Other little fixes.

### CODE REVIEW

File: vexim/userchangesubmit.php [hunk 1]
```diff
@@ -32,21 +32,23 @@
...
+    if (validate_password($_POST['clear'], $_POST['vclear'])) {
+      $cryptedpassword = crypt_password($_POST['clear']);
+      $query = "UPDATE users SET crypt=:crypt WHERE user_id=:user_id";
+      $sth = $dbh->prepare($query);
+      $success = $sth->execute(array(':crypt'=>$cryptedpassword, ':user_id'=>$_SESSION['user_id']));
+      if ($success) {
+        $_SESSION['crypt'] = $cryptedpassword;
+        header ("Location: userchange.php?userupdated");
+        die;
+      } else {
```
Comment (rimas-kudelis) [inline vexim/userchangesubmit.php]: These three lines seem unnecessary, so should probably be removed.

Comment (Udera) [inline vexim/userchangesubmit.php]: Which 3 line out of these 10 lines do you mean?

Comment (rimas-kudelis) [inline vexim/userchangesubmit.php]: Lines 45-47

Comment (Udera) [inline vexim/userchangesubmit.php]: Yes, wrong password makes no sense. Either we remove it or we put a database-error.

Comment (rimas-kudelis) [inline vexim/userchangesubmit.php]: Yeah. I guess we should either use `failupdated=$email_address`, or create a new error message for sql errors.
Or you can just delete these three lines, and let it fail as it does now. :) I still think we have a big rewrite pending anyway.

**Comment by rimas-kudelis** [comment]
Merging this, but I think it could still be improved: correct quota should be calculted from `users.quota` and `domains.quota` on mail delivery, not during mailbox creation. This would allow to leave `users.quota` at zero, which would mean "inherit quota from domain".

**Pull request merged**